### PR TITLE
Add psalm, config provider, empty test, move all configs to .dist

### PR DESCRIPTION
**Pull request by snapshotpl**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | yes
| QA            | no

### Description

- [x] Psalm
- [x] Config provider
- [x] Empty test
- [x] .xml -> .xml.dist
- [x] Migrate PHPUnit config

### CODE REVIEW

File: psalm.xml.dist [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0"?>
+<psalm
+        totallyTyped="true"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns="https://getpsalm.org/schema/config"
+        xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
+    errorBaseline="psalm-baseline.xml"
```
Comment (Ocramius) [inline psalm.xml.dist:7]: Some wonky alignment here, otherwise all good

File: phpunit.xml.dist [hunk 2]
```diff
@@ -1,17 +1,17 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd"
          bootstrap="vendor/autoload.php"
-         colors="true">
```
[the remark this answers is not in the document]
Comment (froschdesign) [reply phpunit.xml.dist:5]: > PHPUnit no longer converts PHP deprecations to exceptions by default (configure `convertDeprecationsToExceptions="true"` to enable this)

Reference:

* https://phpunit.readthedocs.io/en/9.5/configuration.html#the-convertdeprecationstoexceptions-attribute
* https://github.com/sebastianbergmann/phpunit/blob/master/ChangeLog-9.5.md#9510---2021-09-25

[the remark this answers is not in the document]
Comment (snapshotpl) [reply phpunit.xml.dist:5]: Already fixed below

**Review by Ocramius — APPROVED** [review]
:ship: thanks @snapshotpl!

**Pull request merged**